src/classes/RD2_StatusMapper_TEST.cls

```diff
@@ -30,36 +31,290 @@
+    * @return void
+    */
+    public static void assertDefaultStatusValues(Set<String> statusValues) {
+        Boolean hasRequired =
+            statusValues.contains(RD2_Constants.STATUS_ACTIVE)
+            && statusValues.contains(RD2_Constants.STATUS_LAPSED)
+            && statusValues.contains(RD2_Constants.STATUS_CLOSED);
 
-            return this;
+        System.assertEquals(true, hasRequired,
+            'Required Status values should be returned by the mapper: ' + statusValues
+        );
+    }
+
+    /***
+    * @description Verifies all active Status picklist field values are returned by the mapper
+    * @param statusValues Actual Status values returned by the mapper
+    * @return void
+    */
+    public static void assertAllActivePicklistStatusValues(Set<String> statusValues) {
+        Map<String, String> labelByValue = getActiveStatusLabelByValue();
+
+        System.assertEquals(labelByValue.keySet(), statusValues,
+            'All active Status values should be returned by the mapper: ' + statusValues
+        );
+    }
+
+    /**
+     * @description Returns Recurring Donation Status picklist field value and labels
+     * @return Map<String, String> State value mapped by the status key
+     */
+    private static Map<String, String> getActiveStatusLabelByValue() {
+        Map<String, String> labelByValue = new Map<String, String>();
+        List<Schema.PicklistEntry> statusEntries = npe03__Recurring_Donation__c.Status__c.getDescribe().getPicklistValues();
+
+        for (Schema.PicklistEntry entry : statusEntries) {
+            if (entry.isActive()) {
+                labelByValue.put(entry.getValue(), entry.getLabel());
+            }
         }
 
+        return labelByValue;
+    }
+
+    /**
+     * @description Returns default Recurring Donation Status picklist field value and labels
+     * @return Map<String, String> State value mapped by the status key
+     */
+    private static Map<String, String> getDefaultStatusLabelByValue() {
+        return new Map<String, String>{
+            RD2_Constants.STATUS_ACTIVE => RD2_Constants.STATUS_ACTIVE,
+            RD2_Constants.STATUS_LAPSED => RD2_Constants.STATUS_LAPSED,
+            RD2_Constants.STATUS_CLOSED => RD2_Constants.STATUS_CLOSED
+        };
+    }
+
+    /**
+     * @description Builds custom metadatype record
+     * @param status Status value
+     * @return RecurringDonationStatusMapping__mdt
+     */
+    private static RecurringDonationStatusMapping__mdt buildMockCustomMetadataRecord(String status) {
+        return buildMockCustomMetadataRecord(status, status);
+    }
+
+    /**
+     * @description Builds custom metadatype record
+     * @param status Status value
+     * @param state State mapping value
+     * @return RecurringDonationStatusMapping__mdt
+     */
+    private static RecurringDonationStatusMapping__mdt buildMockCustomMetadataRecord(String status, String state) {
+        return new RecurringDonationStatusMapping__mdt(
```

Comment:
It's so nice that these can be mocked with a simple object instance. I wish the same were true with CRLP CMDT's.